module_2/src/com/epam/arrraysofarrays/MatrixAnalyzer.java

```diff
@@ -0,0 +1,139 @@
+        StringBuilder result = new StringBuilder();
+        if (row >= 0 && row < matrix.length && column >= 0 && column < matrix[0].length) {
+            result.append("\nThe numbers of the row [").append(row).append("] are: ");
+            for (int item : matrix[row]) {
+                result.append(item).append(" ");
+            }
+            result.append("\nThe numbers of the column [").append(column).append("] are: ");
+            for (int[] rowArray : matrix) {
+                result.append(rowArray[column]).append(" ");
+            }
+        } else {
+            System.out.println("\nInvalid number of the row or column!");
+        }
+        return result;
+    }
+
+    // task_09
+    public static int[] defineSumOfColumnsItems(int[][] matrix) {
+        int[] columnSums = new int[matrix[0].length];
+        for (int j = 0; j < matrix[0].length; j++) {
+            for (int[] row : matrix) {
+                columnSums[j] += row[j];
+            }
+        }
+        return columnSums;
+    }
+
+    // task_09 searching for max item of the column sums array
+    public static int defineMaxSumItem(int[] columnSums) {
+        int maxSumColumn = 0;
+        for (int i = 1; i < columnSums.length; i++) {
+            if (columnSums[i] > columnSums[maxSumColumn]) {
+                maxSumColumn = i;
+            } else if (columnSums[i] == columnSums[maxSumColumn]) {
+                maxSumColumn = -1;
+            }
+        }
+        return maxSumColumn;
+    }
+
+    // task_10
+    public static StringBuilder definePositiveItemsOfMainDiagonal(int[][] matrix) {
+        StringBuilder positiveNumbersOfMainDiagonal = new StringBuilder();
+        StringBuilder absenceOfPositiveNumbers = new StringBuilder("-1");
+        boolean arePositiveNumbers = false;
+        if (checkIfMatrixIsSquare(matrix)) {
+            for (int i = 0; i < matrix.length; i++) {
+                if (matrix[i][i] > 0) {
+                    positiveNumbersOfMainDiagonal.append(matrix[i][i]).append(" ");
+                    arePositiveNumbers = true;
+                }
+            }
+        } else {
+            System.out.println("\nThe generated matrix isn't square!");
+        }
+        return arePositiveNumbers ? positiveNumbersOfMainDiagonal : absenceOfPositiveNumbers;
+    }
+
+    // task_11
+    public static StringBuilder defineRowsContainedPreassignedNumber(int[][] matrix, int preassignedNumber,
+            int preassignedOccurrences) {
+        StringBuilder numbersOfRows = new StringBuilder();
+        for (int i = 0; i < matrix.length; i++) {
+            int counterOfOccurrences = 0;
+            for (int j = 0; j < matrix[i].length; j++) {
+                if (matrix[i][j] == preassignedNumber) {
+                    counterOfOccurrences++;
+                }
+            }
+            if (counterOfOccurrences >= preassignedOccurrences) {
+                numbersOfRows.append(i).append(" ");
+            }
```

Comment:
I think it will be more effective if you check counterOfOccurrences earlier
```suggestion
                if (matrix[i][j] == preassignedNumber) {
                    if (++counterOfOccurrences >= preassignedOccurrences) {
                        numbersOfRows.append(i).append(" ");
                        break;
                     }
                 }
            }
```